syntax = "proto3";
service Booking {
  rpc CreatBooking(BookingData) returns (BookingData) {}
  rpc DeleteBooking(UserID) returns (BookingData) {}
  rpc GetListBookings(Empty) returns (stream BookingData) {}
  rpc GetBookingByUserID(UserID) returns (stream SingleBookingData) {}
}
message Date {
  string date = 1;
}
message UserID {
  string userid = 1;
}
message DatesMovies{
  string date = 1;
  repeated string movies = 2;
}
message BookingData {
  string userid = 1;
  repeated DatesMovies dates = 2;
}
message SingleBookingData {
  string userid = 1;
  string date = 2;
  string movies = 3;
}
message Empty {
}